File: Exercise1.py
class myStack:
    def __init__(self):
        self.items = []
        print("Initialized Stack")
        pass

    def isEmpty(self):
        return self.items == []
        pass

    def push(self, item):
        print("Adding",item,"to stack")
        return self.items.append(item)
        pass

    def pop(self):
        print("Popping last item")
        if not self.items:
            return None
        #a = self.items[-1]
        #del self.items[-1]
        #return a
        return self.items.pop()
        pass

    def peek(self):
        if not self.items:
            return None
        return self.items[-1]
        pass

    def size(self):
        return len(self.items)
        pass

    def show(self):
        print("printing all values")
        return self.items
        pass

s = myStack()
s.push('1')
s.push('2')
s.push('3')
print("size is ",s.size())

print(s.pop())
print(s.show())
print("size is ",s.size())

print(s.peek())

File: Exercise_1.swift
import Cocoa

// Time Complexity - O(n)
// Space Complexity - O(n)
struct Stack<T> {
    var elements: [T] = []
    
    init() {}
    
    var isEmpty: Bool {
        elements.isEmpty
    }
    
    mutating func push(_ newValue: T) {
        elements.append(newValue)
    }
    
    mutating func pop() -> T? {
        isEmpty ? nil : elements.removeLast()
    }
    
    mutating func peek() -> T? {
        isEmpty ? nil : elements.last
    }
    
    mutating func size() -> Int {
        elements.count
    }
    
    mutating func show() {
       print(String(describing: self))
    }
}

extension Stack: CustomStringConvertible {
    var description: String {
        let topDivider = "------top------\n"
        let bottomDivider = "\n------------"
        let elementsString = elements.map { "\($0)" }.reversed().joined(separator: "\n")
        return topDivider+elementsString+bottomDivider
    }
}

var stackObj = Stack<Int>()
stackObj.push(10)
stackObj.push(20)
stackObj.push(30)

stackObj.show()

print(stackObj.size())

print(stackObj.peek())

print(stackObj.pop())
print(stackObj.pop())

stackObj.show()



